### DISCUSSION TITLE
Pin and Go Returncode

### DISCUSSION THREAD

**Discussion by Slin-nilS** [PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor#347] [Q&A] [answered]
Hi there,

I have a Automation to empty dustbin from the Roborock in the kitchen. The Automation use the "Pin and Go" function. When the bad resident had forgotten to open the door to the kitchen the Roborock park directly behind the door.

How I can send him back to station if he can´t reach the target?

**Comment by PiotrMachowski** [comment, marked as answer]
You can add `vacuum_position` in `attributes` section of Map Extractor config. Then in your automation you can add waiting for vacuum to finish going to target. Add a 10s delay to make yourself sure that map data have been updated and then check vacuum position in attributes of Map Extractor camera. If they are not in a specific range then you can send the vacuum back.

**Reply by Slin-nilS** [reply]
Thanks for the realy fast answer. I will try it. :-)

**Comment by RichiTDI** [comment]
> Hi there,
> 
> I have a Automation to empty dustbin from the Roborock in the kitchen. The Automation use the "Pin and Go" function. When the bad resident had forgotten to open the door to the kitchen the Roborock park directly behind the door.
> 
> How I can send him back to station if he can´t reach the target?

Can you share your automation, please?
